Source/WebCore/platform/audio/SharedAudioDestination.cpp

```diff
@@ -218,11 +210,14 @@ void SharedAudioDestinationAdapter::render(AudioBus* sourceBus, AudioBus* destin
         if (m_needsConfiguration) {
             // The SharedAudioDestinationAdapter avoids allocing or deallocing on the
             // high priority audio thread by merely swapping the contents of the renderer
-            // configuration vectors. After the swap, the contents of m_newRenderers will
-            // be destroyed by configureRenderThread() on the m_configurationWorkQueue.
-            std::swap(m_newRenderers, m_configuredRenderers);
+            // configuration vectors. After the swap, the previous contents of m_configuredRenderers
+            // will be destroyed on the main thread.
+            RenderVector oldRenderers = std::exchange(m_configuredRenderers, WTFMove(m_newRenderers));
             m_needsConfiguration = false;
-            m_configurationSemaphore.signal();
+            callOnMainThread([this, protectedThis = Ref { *this }, oldRenderers = WTFMove(oldRenderers)] () mutable {
+                callAllConfigurationHandlers(true);
+                oldRenderers.clear();
```

Comment:
Surely this is not needed. Seems like a no-op given that it will go out of scope right after and get destroyed anyway.

Reply:
I had thought that if I don't do _something_ here, I'll get an "unused captured variable" error. Let me check to ensure that's not necessary. I also worried about the possibility that an unused-lambda-captured-parameter behaviors like an unused-function-move-parameter and returns to the calling context to be destroyed there. I thought it would just be safer if the vector storage was explicitly destroyed in the main thread.

Reply:
No build errors when I omit it. Seems odd but ::shrug::. I'll remove the explicit `.clear()`.

Reply:
The vector destructor runs so there is a side effect and that's enough to make the compiler happy. You'd get a build error if you passed a raw pointer or a builtin type (like `unsigned`) and didn't use it in the lambda.

Note for example that we capture `protectedThis` in lambdas all the time and rarely use it inside the lambda.